Detectors/EMCAL/simulation/src/RawCreator.cxx

```diff
@@ -0,0 +1,86 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+#include <memory>
+#include <string>
+#include <vector>
+
+#include <boost/program_options.hpp>
+
+#include <TFile.h>
+#include <TTree.h>
+#include <TTreeReader.h>
+
+#include "DataFormatsEMCAL/Digit.h"
+#include "DataFormatsEMCAL/TriggerRecord.h"
+#include "EMCALBase/Geometry.h"
+#include "EMCALSimulation/RawWriter.h"
+
+namespace bpo = boost::program_options;
+
+int main(int argc, const char** argv)
+{
+  bpo::variables_map vm;
+  bpo::options_description opt_general("Usage:\n  " + std::string(argv[0]) +
+                                       " <cmds/options>\n"
+                                       "  Tool will encode emcal raw data from input file\n"
+                                       "Commands / Options");
+  bpo::options_description opt_hidden("");
+  bpo::options_description opt_all;
+  bpo::positional_options_description opt_pos;
+
+  try {
+    auto add_option = opt_general.add_options();
+    add_option("help,h", "Print this help message");
+    add_option("verbose,v", bpo::value<uint32_t>()->default_value(0), "Select verbosity level [0 = no output]");
+    add_option("input-file,i", bpo::value<std::string>()->required(), "Specifies digit input file.");
+    add_option("output-file,o", bpo::value<std::string>()->required(), "Specifies raw ouput file.");
+    add_option("debug,d", bpo::value<uint32_t>()->default_value(0), "Select debug output level [0 = no debug output]");
+
```

Comment:
Could you please add the following options:
1) possibility to produce 1 raw file per link (i.e. FEEID), this is preferred mode for replay using readout.exe. See how it is done e.g. for ITS:
https://github.com/AliceO2Group/AliceO2/blob/b0fc846fae2e7f3b1529ebacdb07c94347331b47/Detectors/ITSMFT/ITS/simulation/src/digi2raw.cxx#L61

2) parsing ``configKeyValues`` options, i.e.
https://github.com/AliceO2Group/AliceO2/blob/b0fc846fae2e7f3b1529ebacdb07c94347331b47/Detectors/ITSMFT/ITS/simulation/src/digi2raw.cxx#L65
+
https://github.com/AliceO2Group/AliceO2/blob/b0fc846fae2e7f3b1529ebacdb07c94347331b47/Detectors/ITSMFT/ITS/simulation/src/digi2raw.cxx#L85
This is necessary for e.g. generation of raw data with custom number of HBFs per TF.

Also, all detectors generate in the output directory a configuration file which can be provided to RawFileReader, this can be done by a simple call: https://github.com/AliceO2Group/AliceO2/blob/b0fc846fae2e7f3b1529ebacdb07c94347331b47/Detectors/ITSMFT/ITS/simulation/src/digi2raw.cxx#L173

Reply:
All requests are added to the RawCreator + RawWriter is adapted.